crates/wit-component/src/builder.rs

```diff
@@ -143,6 +143,7 @@ impl ComponentBuilder {
             ComponentTypeRef::Instance(_) => inc(&mut self.instances),
             ComponentTypeRef::Func(_) => inc(&mut self.funcs),
             ComponentTypeRef::Type(..) => inc(&mut self.types),
+            ComponentTypeRef::Component(_) => inc(&mut self.components),
```

Comment:
I'm not sure of any broader implications of introducing this case. So leaving a note to draw attention to it!

Reply:
Makes sense to me 👍, the builder right now isn't really general-purpose in the sense that it's "just enough" to get `wit-component` working, but definitely no issues in filling out its support of course.